system/core/pages.php

```diff
@@ -347,14 +347,12 @@ function dewikify_link($node) {
 		global $hyphaContentLanguage;
 		global $hyphaPage;
 
-		$uri = $node->getAttribute('href');
-
-		$path = explode('/', $uri, 2);
-		$parts = explode(':', $path[0], 2);
-		if (count($parts) != 2 || $parts[0] != 'hypha')
+		$href = $node->getAttribute('href');
+		if (!preg_match('|^hypha:([^/#]+)(/[^#]*)?(#.*)?$|', $href, $matches))
```

Comment:
Regular expressions are, for a lot of people, hard to read. A comment explaining what it does would be helpful.

Reply:
I do agree. For non-programmers like me, regex are hell.